### PR TITLE
[Docs] Add missing terms in challenge configuration docs

### LINKED ISSUE #3920: Missing terms in challenge configuration docs
The current [docs for challenge configuration] (https://evalai.readthedocs.io/en/latest/configuration.html) do not contain all the terms we allow in [challenge configuration](https://github.com/Cloud-CV/EvalAI-Starters/blob/master/challenge_config.yaml).

It should be fixed asap to help the hosts understand what the terms mean.

Additional next issue (probably slightly harder than this) would be explaning what all is required and what is optional (if anything) in the challenge configuration.

### PR DESCRIPTION
**Pull request by Suryansh5545**
Fixes #3920 Adds the missing terms to challenge configuration docs from Starter challenge config.
Peek 2023-04-05 21-34

@gchhablani  Shall i also do the additional issue improvement (explaining the required and optional fields)in this one? or we need another issue for it?

### CODE REVIEW

**Comment by gchhablani** [comment]
@Suryansh5545 Thanks a lot! This looks good.
Feel free to create another PR for the second issue.

File: docs/source/configuration.md [hunk 1]
```diff
@@ -6,34 +6,28 @@ Following fields are required (and can be customized) in the [`challenge_config.
...
 
-- **is_static_dataset_code_upload**: True/False (specify whether the challenge is static dataset code upload or not. Default is `False`)
-
 - **start_date**: Start DateTime of the challenge (Format: YYYY-MM-DD HH:MM:SS, e.g. 2017-07-07 10:10:10) in `UTC` time zone
 
 - **end_date**: End DateTime of the challenge (Format: YYYY-MM-DD HH:MM:SS, e.g. 2017-07-07 10:10:10) in `UTC` time zone
 
 - **published**: True/False (Boolean field that gives the flexibility to publish the challenge once approved by EvalAI admin. Default is `False`)
 
-- **allowed_email_domains**: A list of domains allowed to participate in the challenge. Leave blank if everyone is allowed to participate. (e.g. `["domain1.com", "domain2.org", "domain3.in"]` Participants with these email domains will only be allowed to participate.)
```
Comment (gchhablani) [inline docs/source/configuration.md:33]: Why were these removed?

**Review by gchhablani — COMMENTED** [review]
Left some comments.

File: docs/source/configuration.md (hunk 1 above)
Comment (Suryansh5545) [reply docs/source/configuration.md:33]: Seeing from the reference config , it seemed that these were removed, but it seems model still have these so i will revert these changes.

**Comment by Suryansh5545** [comment]
@gchhablani  There are some config attributes in ChallengePhaseSplit `((show_leaderboard_by_latest_submission, show_execution_time is_multi_metric_leaderboard)` which are not listed in the example challenge config , So shall i add them.

File: docs/source/configuration.md [hunk 2]
```diff
@@ -6,17 +6,17 @@ Following fields are required (and can be customized) in the [`challenge_config.
...
+- **description**: Long description of the challenge (use a relative path for the HTML file, e.g. `templates/description.html`)
 
-- **evaluation_criteria**: Evaluation criteria and details of the challenge (use a relative path for the HTML file, e.g. `challenge_details/evaluation.html`)
+- **evaluation_details**: Evaluation details and details of the challenge (use a relative path for the HTML file, e.g. `templates/evaluation_details.html`)
 
-- **terms_and_conditions**: Terms and conditions of the challenge (use a relative path for the HTML file, e.g. `challenge_details/tnc.html`)
+- **terms_and_conditions**: Terms and conditions of the challenge (use a relative path for the HTML file, e.g. `templates/terms_and_conditions.html`)
 
-- **image**: Logo of the challenge (use a relative path for the logo in the zip configuration, e.g. `images/logo/challenge_logo.jpg`). **Note**: The image must be in jpg, jpeg or png format.
+- **image**: Logo of the challenge (use a relative path for the logo in the zip configuration, e.g. `./logo.jpg`). **Note**: The image must be in jpg, jpeg or png format.
```
Comment (gchhablani) [inline docs/source/configuration.md:15]: Why was the path here changed?

File: docs/source/configuration.md [hunk 3]
```diff
@@ -6,17 +6,17 @@ Following fields are required (and can be customized) in the [`challenge_config.
...
 
-- **submission_guidelines**: Submission guidelines of the challenge (use a relative path for the HTML file, e.g. `challenge_details/submission_guidelines.html`)
+- **submission_guidelines**: Submission guidelines of the challenge (use a relative path for the HTML file, e.g. `templates/submission_guidelines.html`)
 
-- **evaluation_script**: Python script which will decide how to evaluate submissions in different phases (path of the evaluation script file or folder relative to this YAML file. For e.g. `evaluation_script/`)
+- **evaluation_script**: Python script which will decide how to evaluate submissions in different phases (path of the evaluation script is in root e.g. `./`)
```
Comment (gchhablani) [inline docs/source/configuration.md:19]: Why was the path here changed?

File: docs/source/configuration.md [hunk 4]
```diff
@@ -81,22 +88,58 @@ Following fields are required (and can be customized) in the [`challenge_config.
...
+  - **test_annotation_file**: This file is used for ranking the submission made by a participant. An annotation file can be shared by more than one challenge phase. (Path of the test annotation file relative to this YAML file, e.g. `annotations/test_annotations_devsplit.json`)
 
   - **codename**: Unique id for each challenge phase. Note that the codename of a challenge phase is used to map the results returned by the evaluation script to a particular challenge phase. The codename specified here should match with the codename specified in the evaluation script to perfect mapping.
 
-  - **max_submissions_per_day**: Positive integer which tells the maximum number of submissions per day to a challenge phase.
-
-  - **max_submissions_per_month**: Positive integer which tells the maximum number of submissions per month to a challenge phase.
-
-  - **max_submissions**: a Positive integer that decides the overall maximum number of submissions that can be done to a challenge phase.
+  - **max_submissions_per_day**: Positive integer which tells the maximum number of submissions per day to a challenge phase.(Optional Default value is `100000`)
```
Comment (gchhablani) [inline docs/source/configuration.md:95]: Add comma after `Optional`.

Add space before `(`.

File: docs/source/configuration.md [hunk 5]
```diff
@@ -81,22 +88,58 @@ Following fields are required (and can be customized) in the [`challenge_config.
...
+
+  - **max_submissions_per_month**: Positive integer which tells the maximum number of submissions per month to a challenge phase.(Optional Default value is `100000`)
```
Comment (gchhablani) [inline docs/source/configuration.md:97]: Same here.

File: docs/source/configuration.md [hunk 6]
```diff
@@ -81,22 +88,58 @@ Following fields are required (and can be customized) in the [`challenge_config.
...
+
+  - **max_submissions**: A positive integer that decides the maximum number of total submissions that can be made to the challenge phase.(Optional Default value is `100000`)
```
Comment (gchhablani) [inline docs/source/configuration.md:99]: Same here.

File: docs/source/configuration.md [hunk 7]
```diff
@@ -81,22 +88,58 @@ Following fields are required (and can be customized) in the [`challenge_config.
...
+  - **is_partial_submission_evaluation_enabled**: True/False (indicates whether partial submission evaluation is enabled. Default is `False`)
+  - **allowed_submission_file_types**: This is a list of file types that are allowed for submission(Optional Default is `.json, .zip, .txt, .tsv, .gz, .csv, .h5, .npy`)
 
 - **dataset_splits**:
 
   Dataset splits define the subset of test-set on which the submissions will be evaluated on. Generally, most challenges have two splits:
 
-  1. **test-dev** (Allow participants to make large number of submissions, let them see how they are doing, and let them overfit)
-  2. **test-challenge** (Allow small number of submissions so that they cannot mimic test-set. Use this split to decide the winners for the challenge)
+  1. **train_split** (Allow large number of submissions so participants can train there model)
```
Comment (gchhablani) [inline docs/source/configuration.md:141]: I think the descriptions within the parentheses should not change for both splits.
Also, `train_split` is for evaluation on train split. You can mention `val_split` as validation set, which will be better than `train_split`. I think generally people have validation split for datasets instead of train.

File: docs/source/configuration.md [hunk 8]
```diff
@@ -116,7 +159,7 @@ Following fields are required (and can be customized) in the [`challenge_config.
 
   - **dataset_split_id**: Id of `dataset_split`
 
-  - **visibility**: It will set the visibility of the numbers corresponding to metrics for this `challenge_phase_split`. Select one of the following positive integers based on the visibility level you want:
+  - **visibility**: It will set the visibility of the numbers corresponding to metrics for this `challenge_phase_split`. Select one of the following positive integers based on the visibility level you want:(Optional Default is `3`)
```
Comment (gchhablani) [inline docs/source/configuration.md:162]: Add comma after optional.

Also move the table below here if it renders properly.

File: docs/source/configuration.md [hunk 9]
```diff
@@ -81,22 +88,58 @@ Following fields are required (and can be customized) in the [`challenge_config.
...
+      - name: MultipleChoiceAttribute
+        description: Sample
+        type: checkbox
+        options: ["alpha", "beta", "gamma"]
+      - name: TrueFalseField
+        description: Sample
+        type: boolean
+        required: True
+    ```
+  - **is_restricted_to_select_one_submission**: True/False (indicates whether participants are restricted to submitting only one submission for the challenge. Default is `False`)
```
Comment (gchhablani) [inline docs/source/configuration.md:133]: The description should be 'indicates whether to restrict user to select only one submission for leaderboard'.

File: docs/source/configuration.md [hunk 10]
```diff
@@ -81,22 +88,58 @@ Following fields are required (and can be customized) in the [`challenge_config.
...
+  - **is_partial_submission_evaluation_enabled**: True/False (indicates whether partial submission evaluation is enabled. Default is `False`)
+  - **allowed_submission_file_types**: This is a list of file types that are allowed for submission(Optional Default is `.json, .zip, .txt, .tsv, .gz, .csv, .h5, .npy`)
```
Comment (gchhablani) [inline docs/source/configuration.md:135]: Add space before `(`.

File: docs/source/configuration.md [hunk 11]
```diff
@@ -81,22 +88,58 @@ Following fields are required (and can be customized) in the [`challenge_config.
...
-  - **max_submissions_per_month**: Positive integer which tells the maximum number of submissions per month to a challenge phase.
-
-  - **max_submissions**: a Positive integer that decides the overall maximum number of submissions that can be done to a challenge phase.
+  - **max_submissions_per_day**: Positive integer which tells the maximum number of submissions per day to a challenge phase.(Optional Default value is `100000`)
+
+  - **max_submissions_per_month**: Positive integer which tells the maximum number of submissions per month to a challenge phase.(Optional Default value is `100000`)
+
+  - **max_submissions**: A positive integer that decides the maximum number of total submissions that can be made to the challenge phase.(Optional Default value is `100000`)
+
+  - **default_submission_meta_attributes**: These are the default metadata attributes that will be displayed for all submissions, the metadata attributes are method_name, method_description, project_url, and publication_url.
```
Comment (gchhablani) [inline docs/source/configuration.md:101]: `method_name`, `method_description`, `project_url`, and `publication_url`.

File: docs/source/configuration.md [hunk 12]
```diff
@@ -81,22 +88,58 @@ Following fields are required (and can be customized) in the [`challenge_config.
...
+      - name: method_name
+        is_visible: True
+      - name: method_description
+        is_visible: True
+      - name: project_url
+        is_visible: True
+      - name: publication_url
+        is_visible: True
+      ```
+  - **submission_meta_attributes**: These are the custom metadata attributes that can be added by participants for their submissions. The custom metadata attributes are TextAttribute, SingleOptionAttribute, MultipleChoiceAttribute, and TrueFalseField.
```
Comment (gchhablani) [inline docs/source/configuration.md:113]: `TextAttribute`, `SingleOptionAttribute`, `MultipleChoiceAttribute`, and `TrueFalseField`

**Review by gchhablani — COMMENTED** [review]
Added some comments.

File: docs/source/configuration.md (hunk 2 above)
Comment (Suryansh5545) [reply docs/source/configuration.md:15]: The  path change was due to the location of image in the githiv template, but seeing this now it was only an example of relative path, I will change the path back to original in all locations.

File: docs/source/configuration.md [hunk 13]
```diff
@@ -81,22 +88,59 @@ Following fields are required (and can be customized) in the [`challenge_config.
...
-  - **max_submissions_per_day**: Positive integer which tells the maximum number of submissions per day to a challenge phase.
-
-  - **max_submissions_per_month**: Positive integer which tells the maximum number of submissions per month to a challenge phase.
-
-  - **max_submissions**: a Positive integer that decides the overall maximum number of submissions that can be done to a challenge phase.
+  - **max_submissions_per_day**: Positive integer which tells the maximum number of submissions per day to a challenge phase.( Optional, Default value is `100000`)
+
+  - **max_submissions_per_month**: Positive integer which tells the maximum number of submissions per month to a challenge phase.( Optional, Default value is `100000`)
+
+  - **max_submissions**: A positive integer that decides the maximum number of total submissions that can be made to the challenge phase.( Optional,  Default value is `100000`)
```
Comment (gautamjajoo) [inline docs/source/configuration.md:99]:
```suggestion
  - **max_submissions**: A positive integer that decides the maximum number of total submissions that can be made to the challenge phase. ( Optional,  Default value is `100000`)
```

File: docs/source/configuration.md [hunk 14]
```diff
@@ -81,22 +88,59 @@ Following fields are required (and can be customized) in the [`challenge_config.
...
+  - **is_partial_submission_evaluation_enabled**: True/False (indicates whether partial submission evaluation is enabled. Default is `False`)
+  - **allowed_submission_file_types**: This is a list of file types that are allowed for submission( Optional Default is `.json, .zip, .txt, .tsv, .gz, .csv, .h5, .npy`)
 
 - **dataset_splits**:
 
   Dataset splits define the subset of test-set on which the submissions will be evaluated on. Generally, most challenges have two splits:
 
-  1. **test-dev** (Allow participants to make large number of submissions, let them see how they are doing, and let them overfit)
-  2. **test-challenge** (Allow small number of submissions so that they cannot mimic test-set. Use this split to decide the winners for the challenge)
+  1. **train_split** (Allow participants to make large number of submissions, let them see how they are doing, and let them overfit)
```
Comment (gautamjajoo) [inline docs/source/configuration.md:141]:
```suggestion
  1. **train_split** (Allow participants to make a large number of submissions, let them see how they are doing, and let them overfit)
```

File: docs/source/configuration.md [hunk 15]
```diff
@@ -81,22 +88,59 @@ Following fields are required (and can be customized) in the [`challenge_config.
...
+  2. **test_split** (Allow small number of submissions so that they cannot mimic test_set. Use this split to decide the winners for the challenge)
```
Comment (gautamjajoo) [inline docs/source/configuration.md:142]:
```suggestion
  2. **test_split** (Allow a small number of submissions so that they cannot mimic test_set. Use this split to decide the winners for the challenge)
```

**Review by gautamjajoo — APPROVED** [review]
LGTM!
